db/fetcher/fetcher.go

```diff
@@ -606,16 +621,46 @@ func (df *DocumentFetcher) fetchNext(ctx context.Context) (EncodedDocument, Exec
 			}
 		}
 
+		// Check if we have read access, for document on this collection, with the given identity.
+		if !df.passedPermissionCheck {
+			if !df.acp.HasValue() {
+				// If no acp module, then we have unrestricted access.
+				df.passedPermissionCheck = true
+			} else {
+				hasPermission, err := permission.CheckAccessOfDocOnCollectionWithACP(
+					ctx,
+					df.identity,
+					df.acp.Value(),
+					df.col,
+					acp.ReadPermission,
+					df.kv.Key.DocID,
+				)
+
+				if err != nil {
+					df.passedPermissionCheck = false
```

Comment:
agree! the only place I believe you need it is inside the `processKV` func when it does the reset of the doc reset stuff, which is triggered when we rollover to a new document (which implies new permission checks).

Its also worth noting that when I was implementing the `filter` system inside the `DocumentFetcher`, I originally had a single field like this called `passedFilter` but ran into some problems differentating between a failed filter evaluation, and not yet running the filter. So I added both `passedFilter` and `ranFilter`. Not sure if there are similar issues with this, or if you had any difficulties when determining the access of a doc, but if so feel free to follow this pattern of `passedPermissionCheck` and `ranPermissionCheck` as it might be easier to reason about.

Reply:
The more I think about it/go through the fetcher changes, the more I think this is necessary change. LMK

Reply:
Can likely remove this:
`df.passedPermissionCheck = false`

As it will be false regardless, but IMO I like it being explicitly being false incase in future some other method mutates it for whatever reason. Permission Check should strictly always be false, unless access to document is granted (i.e. it will be true when acp is off).